CONTRIBUTING.md

```diff
@@ -0,0 +1,123 @@
+  review that adds more than 400 lines of new code. (no upper limit on deletions!)
+
+Merging branches and PRs to `main`:
+- The branch author should be the one to merge.
+- Merge PRs via 'Merge Commit' option in GitHub.
+- Delete a branch when you are done with it.
+
+
+# Tests
+
+## Testing Strategy
+
+Keep the test suite
+* **complete**: every feature needs a test.
+* **focused**: test behavior, not implementation.
+* **stable**: tests should be deterministic and fully repeatable.
+* **fast**: no sleeps, no time-based tests.
+
+## CircleCI
+
+All tests in the repo should be executed by CircleCI (see `.circleci/config.yml`)
+automatically upon push to any branch.  (This can be disabled on a per-commit basis
+by appending a **`[ci skip]`** line to the commit message.)
+
+Do not merge branches if CircleCI is not green.
+
+
+# Coding & Documentation Style
+
+## Prettier & ESLint for JavaScript, JSON, & SCSS
+
+All developers are expected to use [prettier](https://prettier.io/) to format their JavaScript or TypeScript scripts in this repository.
+
+> To apply Prettier autoformatting, you can configure your text editor to format on save [according
+> to these docs](https://prettier.io/docs/en/editors.html)
+> - For VS Code, set `editor.formatOnSave` to true in your editor config.
+
+
+ESLint helps identify problematic patterns found in JavaScript code.
+
+> Follow [ESLint Getting Started](https://eslint.org/docs/user-guide/getting-started).
+> To configure ESLint in VS Code, create a `.vscode/settings.json` in the project's root and add
+> ```
+>    "eslint.nodePath": "/home/\${user}/dev/nosh-by-gosh/client/node_modules",
+>    "eslint.workingDirectories": ["client"]
+> ```
+
+
+## Ruff for Python
+
+All developers are expected to use [Ruff](https://github.com/astral-sh/ruff) to lint and format
+their Python code. Ruff is a fast, all-in-one Python linter and formatter that ensures consistency
+and code quality across the project.
+
+First install Ruff:
+
+    pip install ruff
+
+Developers can choose when in their workflow to apply Ruff formatting and linting. Common choices are as a pre-commit hook or format-on-save.
+
+> You may set up Ruff as a pre-commit hook, to automatically run on all staged Python files before each commit:
+> 1. `.pre-commit-config.yaml` is already defined at the top of the repo.
+> 2. Install the pre-commit package: `pip install pre-commit`
+> 3. Install the git hook: `pre-commit install`
+>
+>
+> Or you may configure your text editor to lint and format on save:
+> - For VS Code,
+>   1. Install the "Ruff" extension
+>   2. Add the following to your `settings.json`:
+>      ```
+>      "[python]": {
+>          "editor.formatOnSave": true,
+>          "editor.codeActionsOnSave": {
+>              "source.fixAll": "explicit",
+>              "source.organizeImports": "explicit"
+>          },
+>          "editor.defaultFormatter": "charliermarsh.ruff"
+>      }
+>      ```
```

Comment:
```suggestion
>      ```

Document Python code according to **[NumPy's documentation
standard](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)**.
```

Reply:
unless you have reason to believe that Ruff also  can lint or autoformat docstrings -- in which case we should use that!

Reply:
I have no particular knowledge about Ruff beyond basic code editor usage and what's been introduced in `frizzle-sandbox`, so no such reason on my part :)